tests/workflow_delete_execution.go

```diff
@@ -34,42 +34,42 @@ import (
 	commonpb "go.temporal.io/api/common/v1"
 	enumspb "go.temporal.io/api/enums/v1"
 	"go.temporal.io/api/serviceerror"
-	taskqueuepb "go.temporal.io/api/taskqueue/v1"
 	"go.temporal.io/api/workflowservice/v1"
-	"google.golang.org/protobuf/types/known/durationpb"
+
+	"go.temporal.io/server/common/log/tag"
+	"go.temporal.io/server/common/testing/testvars"
 )
 
 const (
-	waitForPersistence = 10 * time.Second
+	// Internal task processing for DeleteExecutionTask checks if there is no pending CloseExecutionTask
+	// comparing ack levels of the last processed task and DeleteExecutionTask TaskID.
+	// Queue states/ack levels are updated with delay from "history.transferProcessorUpdateAckInterval"
+	// which default 30s, but it is overridden in tests to 1s (see overrideHistoryDynamicConfig in onebox.go).
+	// With few executions closed and deleted in parallel, it is hard to predict time needed for every DeleteExecutionTask
+	// to process. Set it to 20s here, as minimum sufficient interval. Increase it, if tests in this file are failing with
+	// "Condition never satisfied" error.
+	waitForTaskProcessing = 20 * time.Second
```

Comment:
This is a fix :-)

Reply:
Yikes :(

Reply:
wow, it takes more than 10s to verify the task is completed even when UpdateAckInterval is 1s?

Reply:
Yes :-(. Luckly, it is not a problem outside of the test.